### DISCUSSION TITLE
How does detekt automatically registers itself with gradle check?

### DISCUSSION THREAD

**Discussion by root-ansh** [detekt/detekt#4610] [Q&A] [answered]
Hi devs. I find the detekt tool to be very awesome and fun to work with.

My question may sound off topic, but I am currently trying to find  something like "detekt , but for java" (afaik detekt does not work wiith java code, right?)

so far [checkstyle](https://github.com/checkstyle/checkstyle) seems to work but it doesn't automatically registers for check task as detekt plugin does. 

so i was wondering if i could add any code in _my app_ to make a single `gradle check` command that would perform analysis on both kotlin and java code? 

many thanks

**Comment by marschwar** [comment, marked as answer]
You are right, your question is a little off topic ;). 

> afaik detekt does not work wiith java code, right?

That is correct. `detekt` is a static code analyzer for kotlin only.

> so far [checkstyle](https://github.com/checkstyle/checkstyle) seems to work but it doesn't automatically registers for check task as detekt plugin does.

That I cannot believe. If the checkstyle (and spotbugs as well) plugin is applied to the project, it is automatically added as a check task dependency.

[edit] Please check out [this test in the gradle project](https://github.com/gradle/gradle/blob/be0dac3e960c8ac5f49be6933bb7c323f597852b/subprojects/code-quality/src/test/groovy/org/gradle/api/plugins/quality/CheckstylePluginTest.groovy#L112-L122)

> so i was wondering if i could add any code in my app to make a single gradle check command that would perform analysis on both kotlin and java code?

What you could do is add any task as a dependency of the check task like this.

```
tasks.named("check").configure {
  dependsOn myCustomCheckTask
}
```

**Reply by root-ansh** [reply]
hi thanks for replying. the dependency block worked! for me , i had a custom task like this :
```
task Checkstyle(type: Checkstyle) {
    source 'src/main/java'
    ignoreFailures true
    showViolations true
    include '**/*.java'
    classpath = files()
}
```

it worked when i run the command `gradle checkstyle` but was not running via `gradle check` . so after adding :

```
tasks.named("check").configure {
    dependsOn "Checkstyle" // i was previously trying "checkstyle" in smallcase , but that didn't worked. 
}

```
it worked correctly.
